stdlib/public/core/MemoryLayout.swift

```diff
@@ -11,60 +11,157 @@
+/// static properties of `MemoryLayout<Point>`.
+///
+///     // MemoryLayout<Point>.size == 17
+///     // MemoryLayout<Point>.stride == 24
+///     // MemoryLayout<Point>.alignment == 8
+///
+/// Always use a multiple of a type's `stride` instead of its `size` when
+/// allocating memory or accounting for the distance between instances in
+/// memory. This example allocates untyped, uninitialized memory with space
+/// for four instances of `Point`.
+///
+///     let count = 4
+///     let pointPointer = UnsafeMutableRawPointer.allocate(
+///             bytes: count * MemoryLayout<Point>.stride,
+///             alignedTo: MemoryLayout<Point>.alignment)
 public enum MemoryLayout<T> {
-  /// The contiguous memory footprint of `T`.
+  /// The contiguous memory footprint of `T`, in bytes.
   ///
-  /// Does not include any dynamically-allocated or "remote" storage. In
-  /// particular, `MemoryLayout<T>.size`, when `T` is a class type, is the same
-  /// regardless of how many stored properties `T` has.
+  /// A type's size does not include any dynamically allocated or remote
+  /// storage. In particular, `MemoryLayout<T>.size`, when `T` is a class
+  /// type, is the same regardless of how many stored properties `T` has.
+  ///
+  /// When allocating memory for multiple instances of `T` using an unsafe
+  /// pointer, use a multiple of the type's `stride` property instead of its
+  /// `size`.
+  ///
+  /// - SeeAlso: `stride`
   @_transparent
   public static var size: Int {
     return Int(Builtin.sizeof(T.self))
   }
 
   /// The number of bytes from the start of one instance of `T` to the start of
-  /// the next in an `Array<T>`.
+  /// the next when stored in contiguous memory or in an `Array<T>`.
   ///
   /// This is the same as the number of bytes moved when an `UnsafePointer<T>`
-  /// is incremented. `T` may have a lower minimal alignment that trades runtime
-  /// performance for space efficiency. The result is always positive.
+  /// instance is incremented. `T` may have a lower minimal alignment that
+  /// trades runtime performance for space efficiency. This value is always
+  /// positive.
   @_transparent
   public static var stride: Int {
     return Int(Builtin.strideof(T.self))
   }
 
-  /// The default memory alignment of `T`.
+  /// The default memory alignment of `T`, in bytes.
+  ///
+  /// Use the `alignment` property for a type when allocating memory using an
+  /// unsafe pointer. This value is always positive.
   @_transparent
   public static var alignment: Int {
     return Int(Builtin.alignof(T.self))
   }
 }
 
 extension MemoryLayout {
-  /// Returns the contiguous memory footprint of `T`.
+  /// Returns the contiguous memory footprint of the given instance.
+  ///
+  /// The result does not include any dynamically allocated or remote
+  /// storage. In particular, `MemoryLayout.size(ofValue: x)`, when `x` is an
+  /// instance of a class `C`, is the same regardless of how many stored
+  /// properties `C` has.
```

Comment:
To me this is a bit weird. It is skirting around a class being a pointer. Why not just call that out in a subsequent sentence? Perhaps use the term "out of line storage"? Just a thought.